cmd/http/tracer.go

```diff
@@ -24,6 +32,39 @@ func tracerProvider(url string) (*trace.TracerProvider, error) {
 			semconv.ServiceNameKey.String(config.TraceName),
 			attribute.String("environment", config.TraceEnv),
 		)),
+		trace.WithSampler(sampler),
 	)
 	return tp, nil
 }
+
+type ProbabilisticSampler struct {
+	max      int
+	boundary int // values equal or greater than boundary would be dropped; 1/00 prob can be expressed as max=100, boundary=1
+}
+
+func (s ProbabilisticSampler) ShouldSample(p trace.SamplingParameters) trace.SamplingResult {
+	r := rand.Intn(s.max)
+	if r >= s.boundary {
+		return trace.SamplingResult{
+			Decision:   trace.Drop,
+			Tracestate: oteltrace.SpanContextFromContext(p.ParentContext).TraceState(),
+		}
+	}
+
+	return trace.SamplingResult{
+		Decision:   trace.RecordAndSample,
+		Tracestate: oteltrace.SpanContextFromContext(p.ParentContext).TraceState(),
+	}
+}
+
+func (s ProbabilisticSampler) Description() string {
+	return "probabilistic sampler"
+}
+
+func NewProbabilisticSampler(probability float64) *ProbabilisticSampler {
+	const max = 1000.0
+	boundary := max * probability
+	return &ProbabilisticSampler{max, int(boundary)}
+}
+
+var _ trace.Sampler = NewProbabilisticSampler(0.001)
```

Comment:
0.001 to be derived from max (1000 now)

Reply:
this is not a real code but just golang's way to ensure the interface of Sampler is implemented.

Reply:
I mean "const max = 1000.0"

Reply:
I see; for now let's use 1000 as the base to allow 1/1000 as the minimum sampling. in the future if we need to enable finer granular sampler like 1/1M, we can change that value later. How do you think?

Reply:
It shall be fine. But I suggest it might be better to add some comments here since other developer might find that 1/10000 does not work and they have to go through codes to find out.

Reply:
used math.MaxInt32 as the base to avoid issue that might arise from less than 1/1000